front/lib/actions/mcp_internal_actions/servers/ashby/index.ts

```diff
@@ -215,6 +216,241 @@ function createServer(
+          return new Err(
+            new MCPError(
+              `No submitted interview feedback found for candidate ${candidate.name}.`
+            )
+          );
+        }
+
+        allFeedback.sort((a, b) => {
+          const dateA = new Date(a.submittedAt!);
+          const dateB = new Date(b.submittedAt!);
+          return dateB.getTime() - dateA.getTime();
+        });
+
+        const recapText = renderInterviewFeedbackRecap(candidate, allFeedback);
+
+        return new Ok([
+          {
+            type: "text" as const,
+            text: recapText,
+          },
+        ]);
+      }
+    )
+  );
+
+  server.tool(
+    "create_candidate_note",
+    "Create a note on a candidate's profile in Ashby. " +
+      "The note content can include basic HTML formatting (supported tags: h1-h6, p, b, i, u, a, ul, ol, li, code, pre).",
+    {
+      email: z
+        .string()
+        .optional()
+        .describe(
+          "Email address of the candidate (partial matches supported)."
+        ),
+      name: z
+        .string()
+        .optional()
+        .describe("Name of the candidate (partial matches supported)."),
+      noteContent: z
+        .string()
+        .describe("The content of the note in HTML format."),
+    },
+    withToolLogging(
+      auth,
+      {
+        toolNameForMonitoring: "ashby_create_candidate_note",
+        agentLoopContext,
+      },
+      async ({ email, name, noteContent }) => {
+        if (!email && !name) {
+          return new Err(
+            new MCPError(
+              "At least one search parameter (email or name) must be provided."
+            )
+          );
+        }
+
+        const apiKeyResult = await getAshbyApiKey(auth, agentLoopContext);
+        if (apiKeyResult.isErr()) {
+          return new Err(apiKeyResult.error);
+        }
+
+        const client = new AshbyClient(apiKeyResult.value);
+
+        const searchResult = await client.searchCandidates({ email, name });
+        if (searchResult.isErr()) {
+          return new Err(
+            new MCPError(
+              `Failed to search candidates: ${searchResult.error.message}`
+            )
+          );
+        }
+
+        const candidates = searchResult.value.results;
+        if (candidates.length === 0) {
+          return new Ok([
+            {
+              type: "text" as const,
+              text: "No candidates found matching the search criteria.",
+            },
+          ]);
+        }
+
+        if (candidates.length > 1) {
+          const candidatesList = candidates
+            .map(
+              (c, i) =>
+                `${i + 1}. ${c.name} (${c.primaryEmailAddress?.value ?? "no email"})`
+            )
+            .join("\n");
+          return new Err(
+            new MCPError(
+              `Multiple candidates found. Please refine your search:\n\n${candidatesList}`
+            )
+          );
+        }
+
+        const candidate = candidates[0];
```

Comment:
you could extract this in a helper function as it's the same as above. Something like `getCandidate` or something